nightfall-deployer/contracts/Challenges.sol

```diff
@@ -122,100 +137,101 @@ contract Challenges is Stateful, Key_Registry, Config {
+        state.areBlockAndTransactionReal(
+            transaction1.blockL2,
+            transaction1.transaction,
+            transaction1.transactionIndex,
+            transaction1.transactionSiblingPath
+        );
+        state.areBlockAndTransactionReal(
+            transaction2.blockL2,
+            transaction2.transaction,
+            transaction2.transactionIndex,
+            transaction2.transactionSiblingPath
+        );
+
+        require(
+            transaction1.blockL2.blockNumberL2 != transaction2.blockL2.blockNumberL2,
+            'Cannot be the same block'
+        );
+
         ChallengesUtil.libChallengeNullifier(
-            transactions1[transaction1Index],
+            transaction1.transaction,
             nullifier1Index,
-            transactions2[transaction2Index],
+            transaction2.transaction,
             nullifier2Index
         );
-        // If the duplicate exists in the same block, the index cannot be the same
-        if (block1.blockNumberL2 == block2.blockNumberL2)
-            require(transaction1Index != transaction2Index, 'Cannot be the same index');
-        // first, check we have real, in-train, contiguous blocks
-        state.areBlockAndTransactionsReal(block1, transactions1);
-        state.areBlockAndTransactionsReal(block2, transactions2);
+
         // Delete the latest block of the two
-        if (block1.blockNumberL2 > block2.blockNumberL2) {
-            challengeAccepted(block1);
+        if (transaction1.blockL2.blockNumberL2 > transaction2.blockL2.blockNumberL2) {
+            challengeAccepted(transaction1.blockL2);
         } else {
-            challengeAccepted(block2);
+            challengeAccepted(transaction2.blockL2);
         }
     }
 
     function challengeProofVerification(
-        Block calldata blockL2,
-        Transaction[] calldata transactions,
-        uint256 transactionIndex,
+        TransactionInfoBlock calldata transaction,
         Block[4] calldata blockL2ContainingHistoricRoot,
-        Transaction[][4] calldata transactionsOfblockL2ContainingHistoricRoot,
         uint256[8] memory uncompressedProof,
         bytes32 salt
     ) external onlyBootChallenger {
         checkCommit(msg.data);
-        state.areBlockAndTransactionsReal(blockL2, transactions);
+        state.areBlockAndTransactionReal(
+            transaction.blockL2,
+            transaction.transaction,
+            transaction.transactionIndex,
+            transaction.transactionSiblingPath
+        );
 
         PublicInputs memory extraPublicInputs = PublicInputs(
             [uint256(0), 0, 0, 0],
             super.getMaticAddress()
         );
 
-        if (uint256(transactions[transactionIndex].nullifiers[0]) != 0) {
-            state.areBlockAndTransactionsReal(
-                blockL2ContainingHistoricRoot[0],
-                transactionsOfblockL2ContainingHistoricRoot[0]
-            );
+        if (uint256(transaction.transaction.nullifiers[0]) != 0) {
```

Comment:
Can this sequence of checks just be compressed into a `for loop` that goes to `nullifiers.length`